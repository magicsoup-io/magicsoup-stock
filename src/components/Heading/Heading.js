import styled from 'styled-components'

import Text from '../Text'

const themed = key => props => props.theme[key]

const Heading = styled(Text)`
  ${themed('Heading')}
`
/** @component */
export default Heading

Heading.defaultProps = {
  as: 'h2',
  m: 0,
  fontSize: 4,
  fontWeight: 'bold',
}
