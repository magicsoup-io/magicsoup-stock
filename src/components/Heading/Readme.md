Heading:

```jsx
<Heading as="h1" fontSize={8} color="primary">Say hello</Heading>

<Heading color="primary">Say hello</Heading>
```
