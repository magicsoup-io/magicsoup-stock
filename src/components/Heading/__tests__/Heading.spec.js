import React from 'react'
import renderer from 'react-test-renderer'

import Heading from '../Heading'

it('renders <Heading /> without throwing', () => {
  const tree = renderer.create(<Heading> I am a Heading</Heading>).toJSON()
  expect(tree).toMatchSnapshot()
})
