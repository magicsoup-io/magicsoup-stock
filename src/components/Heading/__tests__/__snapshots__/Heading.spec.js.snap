// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Heading /> without throwing 1`] = `
<h2
  className="sc-bdVaJa sc-bwzfXH sc-htpNat ikroEO"
  fontSize={4}
  fontWeight="bold"
>
   I am a Heading
</h2>
`;
